File: Python3/225. Implement Stack using Queues.py
class MyStack:

    def __init__(self):
        self.q = MQueue()

    def push(self, x: int) -> None:
        s = self.q.size()
        self.q.push(x)
        
        while s:
            self.q.push(self.q.pop())
            s -= 1
        
    def pop(self) -> int:
        return self.q.pop()

    def top(self) -> int:
        return self.q.top()

    def empty(self) -> bool:
        return self.q.empty()


# Your MyStack object will be instantiated and called as such:
# obj = MyStack()
# obj.push(x)
# param_2 = obj.pop()
# param_3 = obj.top()
# param_4 = obj.empty()

class MQueue:
    
    def __init__(self):
        self.queue = []
        self.s = 0
        
    def push(self, x: int) -> None:
        self.queue.append(x)
        self.s += 1

    def pop(self) -> int:
        self.s -= 1
        return self.queue.pop(0)

    def top(self) -> int:
        return self.queue[0]

    def empty(self) -> bool:
        return len(self.queue)==0
    
    def size(self) -> int:
        return self.s

File: Java/225. Implement Stack using Queues.java
class MyStack {
    
    private Queue<Integer> q;
    
    public MyStack() {
        q = new LinkedList<>();
    }
    
    public void push(int x) {
        int s = q.size();
        q.offer(x);
        while(s > 0){
            q.offer(q.poll());
            s--;
        }
    }
    
    public int pop() {
        return q.poll();
    }
    
    public int top() {
        return q.peek();
    }
    
    public boolean empty() {
        return q.isEmpty();
    }
}

/**
 * Your MyStack object will be instantiated and called as such:
 * MyStack obj = new MyStack();
 * obj.push(x);
 * int param_2 = obj.pop();
 * int param_3 = obj.top();
 * boolean param_4 = obj.empty();
 */